ts/screens/wallet/payment/ScanQrCodeScreen.tsx

```diff
@@ -304,66 +300,33 @@ class ScanQrCodeScreen extends React.Component<Props, State> {
             backgroundColor={customVariables.colorWhite}
           />
           <ScrollView bounces={false}>
-            {this.state.isFocused && this.state.permissionRationaleDisplayed && (
-              <QRCodeScanner
-                onRead={(reading: { data: string }) =>
-                  this.onQrCodeData(reading.data)
-                }
-                ref={this.qrCodeScanner}
-                containerStyle={styles.cameraContainer as any}
-                showMarker={true}
-                cameraStyle={styles.camera as any}
-                customMarker={
-                  <CameraMarker
-                    screenWidth={screenWidth}
-                    state={this.state.scanningState}
-                  />
-                }
-                bottomContent={
-                  <View>
-                    <ButtonDefaultOpacity
-                      onPress={this.onShowImagePicker}
-                      style={styles.button}
-                      bordered={true}
-                    >
-                      <Text>{I18n.t("wallet.QRtoPay.chooser")}</Text>
-                    </ButtonDefaultOpacity>
-                    <View style={styles.content}>
-                      <View spacer={true} />
-                      <Text style={[styles.padded, styles.bottomText]}>
-                        {I18n.t("wallet.QRtoPay.cameraUsageInfo")}
-                      </Text>
-                      <View spacer={true} extralarge={true} />
-                    </View>
-                  </View>
-                }
-                // "captureAudio" enable/disable microphone permission
-                cameraProps={{ captureAudio: false }}
-                // "checkAndroid6Permissions" property enables permission checking for
-                // Android versions greater than 6.0 (23+).
-                checkAndroid6Permissions={true}
-                // "notAuthorizedView" is by default available on iOS systems ONLY.
-                // In order to make Android systems act the same as iOSs you MUST
-                // enable "checkAndroid6Permissions" property as well.
-                // On devices before SDK version 23, the permissions are automatically
-                // granted if they appear in the manifest, so message customization would
-                // be impossible.
-                notAuthorizedView={
-                  <View style={styles.notAuthorizedContainer}>
-                    <Text style={styles.notAuthorizedText}>
-                      {I18n.t("wallet.QRtoPay.enroll_cta")}
-                    </Text>
+            <BarcodeCamera
+              onBarcodeScanned={this.handleBarcodeScanned}
+              disabled={!this.state.isFocused}
+              marker={
+                <CameraMarker
+                  screenWidth={screenWidth}
+                  state={this.state.scanningState}
```

Comment:
We should keep the scheduled timeout to restore the scanning state when a not valid QRCode is scanned

### previous code
```typescript
this.scannerReactivateTimeoutHandler = setTimeout(() => {
      // eslint-disable-next-line
      this.scannerReactivateTimeoutHandler = undefined;
      if (this.qrCodeScanner.current) {
        this.qrCodeScanner.current.reactivate();
        this.setState({
          scanningState: "SCANNING"
        });
      }
    }, QRCODE_SCANNER_REACTIVATION_TIME_MS);
```

### before
https://user-images.githubusercontent.com/822471/166941468-428150a5-0c9a-4fa5-92b4-12e2de75e38b.mp4


### now
https://user-images.githubusercontent.com/822471/166941417-6bed1088-7827-4be8-b7e9-8a887419e777.mp4

Reply:
This functionality has been restored in this commit: https://github.com/pagopa/io-app/pull/3928/commits/b0c47a7c022de4f543d051567653d397a72084a3 ✌️